### ISSUE TITLE
[BUG] BlobClient.exists() is greedy for heap space

### ISSUE THREAD

**Issue by johnwalkera4o** [Azure/azure-sdk-for-java#31555]
**Describe the bug**
Calling BlobClient.exists() seems to require a significant amount of heap space, proportional to the size of the blob.

**To Reproduce**
Manually upload object to blob storage. Update the code snippet below with the URL to the blob and the managed resource ID. Run setting the -Xmx flag. For a 250MB blob the max heap size has to be set to ~450MB, for a 1.3GB blob we seem to require almost 2GB heap size otherwise the exists call hangs forever. In the snippet below I've used existsWithResponse in order to set a 30s timeout to avoid getting stuck on the exists call. If the max heap size is set too low then we'll hit the IllegalStateException after the timeout duration.

Calling .exists() for a blob that does not exist always returns quickly and requires a very small max heap size - 150MB is fine.

***Code Snippet***

    import java.time.Duration;
    import com.azure.core.http.rest.Response;
    import com.azure.identity.ManagedIdentityCredential;
    import com.azure.identity.ManagedIdentityCredentialBuilder;
    import com.azure.storage.blob.BlobClient;
    import com.azure.storage.blob.BlobClientBuilder;

    public class AzureExistsTest {

        public static void main(String[] args) throws Exception
        {
            String blobURI = "https://container.blob.core.windows.net/<path to blob>";
            String resourceId = "<RESOURCE ID>";
            ManagedIdentityCredential managedIdentityCredential =
                  new ManagedIdentityCredentialBuilder().resourceId(resourceId).build();

            BlobClient blobClient = new BlobClientBuilder().credential(managedIdentityCredential)
                                                           .endpoint(blobURI)
                                                           .buildClient();
            Boolean blobExists = false;

            try
            {
                Response<Boolean> lresponse = blobClient.existsWithResponse(Duration.ofSeconds(30), null);
                blobExists = lresponse.getValue();
                System.out.println("Blob exists returned: " + blobExists);
            }
            catch (IllegalStateException ex)
            {
                System.out.println("Illegal state exception: " + ex.getMessage());
            }
            catch (Exception ex)
            {
                System.out.println("Caught exception: " + ex.getMessage());
            }
        }
    }

**Expected behavior**
Expect BlobClient.exists() to return within a few seconds for blobs of any size with a true/false response whilst still being able to restrict the maximum heap size of the overall Java process to less than 1 GB.

**Setup (please complete the following information):**
 - OS: CentOS
 - IDE: None
 - Library/Libraries: com.azure:azure-core:1.32.0 
                               com.azure:azure-storage-blob:12.19.1
                               com.azure:azure-identity:1.5.5
                               com.azure:azure-storage-common:12.18.1
                               com.azure:azure-storage-internal-avro:12.4.2
                               com.azure:azure-core-http-netty:1.12.5  
 - Java version: 8
 - App Server/Environment: None
 - Frameworks: None

**Information Checklist**
Kindly make sure that you have added all the following information above and checkoff the required fields otherwise we will treat the issuer as an incomplete report
- [x] Bug Description Added
- [x] Repro Steps Added
- [x] Setup information Added

**Comment by joshfree** [comment]
@jaschrep-msft could you follow up with @johnwalkera4o on this issue

**Comment by ibrahimrabab** [comment]
Hi @johnwalkera4o 
Thank you for bringing up this issue. We discovered it was due to the logic in our backend where void-based responses would eagerly be read and could result in error cases. Depending on size of the blob, buffers were created in preparation to read the response into memory eagerly. This explains why we were seeing excessive memory usage or OutOfMemoryErrors, depending on the size of the blob or file. 

Now we just put out a fix so that void-based responses have special handling where the network response will be ignored and drained instead of attempting to read them, thus resolving the issue of excessive memory usage. 

Thanks again for bringing this to our attention! Hopefully this fixes the issues you've been seeing.
